app/scripts/filters/resources.js

```diff
@@ -1029,32 +1029,66 @@ angular.module('openshiftConsole')
       }
 
-      var reason = pod.status.reason || pod.status.phase;
+      var initializing = false;
+      var reason;
 
-      // Print detailed container reasons if available. Only the last will be
-      // displayed if multiple containers have this detail.
-      angular.forEach(pod.status.containerStatuses, function(containerStatus) {
-        var containerReason = _.get(containerStatus, 'state.waiting.reason') || _.get(containerStatus, 'state.terminated.reason'),
-            signal,
-            exitCode;
+      _.each(pod.status.initContainerStatuses, function(initContainerStatus) {
+        var initContainerState = _.get(initContainerStatus, 'state');
 
-        if (containerReason) {
-          reason = containerReason;
+        if (initContainerState.terminated && initContainerState.terminated.exitCode === 0) {
+          // initialization is complete
           return;
         }
 
-        signal = _.get(containerStatus, 'state.terminated.signal');
-        if (signal) {
-          reason = "Signal: " + signal;
-          return;
+        if (initContainerState.terminated) {
+          // initialization is failed
+          if (!initContainerState.terminated.reason) {
+            if (initContainerState.terminated.signal) {
+              reason = "Init Signal: " + initContainerState.terminated.signal;
+            } else {
+              reason = "Init Exit Code: " + initContainerState.terminated.exitCode;
+            }
+          } else {
+            reason = "Init " + initContainerState.terminated.reason;
+          }
+          initializing = true;
+          return true;
         }
 
-        exitCode = _.get(containerStatus, 'state.terminated.exitCode');
-        if (exitCode) {
-          reason = "Exit Code: " + exitCode;
+        if (initContainerState.waiting && initContainerState.waiting.reason && initContainerState.waiting.reason !== 'PodInitializing') {
+          reason = "Init " + initContainerState.waiting.reason;
+          initializing = true;
         }
       });
 
+      if (!initializing) {
+        reason = pod.status.reason || pod.status.phase;
+
+        // Print detailed container reasons if available. Only the last will be
+        // displayed if multiple containers have this detail.
+        _.each(pod.status.containerStatuses, function(containerStatus) {
+          var containerReason = _.get(containerStatus, 'state.waiting.reason') || _.get(containerStatus, 'state.terminated.reason'),
+              signal,
+              exitCode;
+
+          if (containerReason) {
+            reason = containerReason;
+            return true;
+          }
+
+          signal = _.get(containerStatus, 'state.terminated.signal');
+          if (signal) {
+            reason = "Signal: " + signal;
+            return true;
+          }
+
+          exitCode = _.get(containerStatus, 'state.terminated.exitCode');
+          if (exitCode) {
+            reason = "Exit Code: " + exitCode;
```

Comment:
We should `return true` after this line, too, so we stop looking at other containers like the other checks above. Otherwise it'll keep iterating.